scenarios/throughput_stress.go

```diff
@@ -288,7 +254,30 @@ func (t *tpsExecutor) Run(ctx context.Context, info loadgen.ScenarioInfo) error
 		},
 		completedWorkflows,
 		t.config.VisibilityVerificationTimeout,
-	)
+	); err != nil {
+		return err
+	}
+
+	// Post-scenario: ensure there are no failed or terminated workflows for this run.
+	for _, status := range []enums.WorkflowExecutionStatus{
+		enums.WORKFLOW_EXECUTION_STATUS_TERMINATED, enums.WORKFLOW_EXECUTION_STATUS_FAILED,
+	} {
+		statusQuery := fmt.Sprintf(
+			"%s='%s' and ExecutionStatus = '%s'",
+			ThroughputStressScenarioIdSearchAttribute, info.RunID, status)
+		visibilityCount, err := info.Client.CountWorkflow(ctx, &workflowservice.CountWorkflowExecutionsRequest{
+			Namespace: info.Namespace,
+			Query:     statusQuery,
+		})
+		if err != nil {
+			return fmt.Errorf("failed to run query %q: %w", statusQuery, err)
+		}
+		if visibilityCount.Count > 0 {
+			return fmt.Errorf("unexpected %d workflows with status %s", visibilityCount.Count, status)
+		}
+	}
```

Comment:
Added this ^ as another sanity check since the `MinVisibilityCountEventually` isn't 100% exact (ie "min")